src/MoBi.Core/OSPSuite.Core/Domain/Builder/CoreExpressionProfile.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections;
+using OSPSuite.Utility.Collections;
+using System.Collections.Generic;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility.Extensions;
+using static OSPSuite.Core.CoreConstants.ContainerName;
+
+namespace OSPSuite.Core.Domain.Builder
+{
+   public class CoreExpressionProfile : BuildingBlock, IEnumerable<ExpressionParameter>
+   {
+      private readonly ICache<IObjectPath, ExpressionParameter> _allValues;
+      public CoreExpressionProfile()
+      {
+         _allValues = new Cache<IObjectPath, ExpressionParameter>(x => x.Path, x => null);
+      }
+
+      public virtual string MoleculeName { set; get; }
+
+      public string Species { set; get; }
+
+      public ExpressionType Type { set; get; }
+
+      public int PKSimVersion { set; get; }
+
+      public virtual string Category { set; get; }
+
+      public override string Name
+      {
+         get => ExpressionProfileName(MoleculeName, Species, Category);
+         set
+         {
+            if (string.Equals(Name, value))
+            {
+               return;
+            }
+
+            var (moleculeName, _, category) = NamesFromExpressionProfileName(value);
```

Comment:
probably MoleculeName should be assigned too. Thisis not done in PKSim because it's the name of the molecule but needs to be done here. (Molecule private set same as Category)